java/src/Ice/src/main/java/com/zeroc/Ice/Network.java

```diff
@@ -698,11 +698,10 @@ public static java.util.ArrayList<java.net.InetAddress> getLocalAddresses(
     }
 
     public static java.util.ArrayList<String> getHostsForEndpointExpand(
-            String host, int protocolSupport, boolean includeLoopback) {
+            String host, int protocolSupport) {
         java.util.ArrayList<String> hosts = new java.util.ArrayList<>();
         if (isWildcard(host)) {
-            for (java.net.InetAddress addr :
-                    getLocalAddresses(protocolSupport, includeLoopback, false)) {
+            for (java.net.InetAddress addr : getLocalAddresses(protocolSupport, true, false)) {
```

Comment:
I was looking why we have to keep the include loopback in Java unlike in other mappings.

Seems we only set this flag to false in https://github.com/zeroc-ice/ice/blob/55308b51034a21c7f5ccf3c8c1ef830c3476c444/java/src/Ice/src/main/java/com/zeroc/Ice/Network.java#L565

This overload is only used by the syslog logger to get the localhost address

https://github.com/zeroc-ice/ice/blob/55308b51034a21c7f5ccf3c8c1ef830c3476c444/java/src/Ice/src/main/java/com/zeroc/Ice/SysLoggerI.java#L75

It would make more sense to use a property to configure the syslog daemon, with a default. Then we can get rid of this  getLocalAddress overload and the include loopback flag.

For a separate issue

Reply:
See #2841.